doc/release/1.11.0-notes.rst

```diff
@@ -41,8 +41,21 @@ Future Changes
 The following changes are scheduled for Numpy 1.12.0.
 
 * Support for Python 2.6, 3.2, and 3.3 will be dropped.
-* Slicing a ``MaskedArray`` will return views of both data **and** mask.
-  Currently the mask is returned as a copy.
+* When setting items of a ``MaskedArray`` using the square bracket
+  notation NumPy will in the future never copy the mask. Some
+  examples of this are ``child = ma_array[::2]; view[3] = 3``, where
+  the mask of ``child`` is a view into the mask of ``ma_array``.
+  Also when creating ``ma_array = MaskedArray(data, mask)`` will
+  make ``ma_array`` view the data in ``mask`` and setting an item
+  will currently copy ``mask`` but in the future modify the original.
```

Comment:
Could just document that unshare_mask is unreliable after this, I can't
imagine people will want to intentionally be using both. Could also make at
least the new version of unshare_mask copy unconditionally (or maybe it
already does this) in preparation for removing _sharedmask logic entirely.
On Feb 28, 2016 11:53, "seberg" <notifications@github.com> wrote:

> In doc/release/1.11.0-notes.rst
> <https://github.com/numpy/numpy/pull/7363#discussion_r54355225>:
>
> > @@ -41,8 +41,21 @@ Future Changes
> >  The following changes are scheduled for Numpy 1.12.0.
> >
> >  * Support for Python 2.6, 3.2, and 3.3 will be dropped.
> > -* Slicing a ``MaskedArray`` will return views of both data **and** mask.
> > -  Currently the mask is returned as a copy.
> > +* When setting items of a ``MaskedArray`` using the square bracket
> > +  notation NumPy will in the future never copy the mask. Some
> > +  examples of this are ``child = ma_array[::2]; view[3] = 3``, where
> > +  the mask of ``child`` is a view into the mask of ``ma_array``.
> > +  Also when creating ``ma_array = MaskedArray(data, mask)`` will
> > +  make ``ma_array`` view the data in ``mask`` and setting an item
> > +  will currently copy ``mask`` but in the future modify the original.
>
> Yeah, that logic was there since a very long time (probably forever). but
> setting _sharedmask to false, might break explicit calls to unshare_mask,
> hmmpf, not hat I assume there are many, but still.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/numpy/numpy/pull/7363/files#r54355225>.
>